File: Day179_2025/Q179_GFG.cpp
class Solution {
  public:
    vector<int> countLessEq(vector<int>& a, vector<int>& b) {
        // Q. Counting elements in two arrays
        // code here
        sort(b.begin(),b.end());
        vector<int>ans;
        for(auto it:a){
            int start = 0;
            int end = b.size() - 1;
            int mid = start + (end - start) / 2;
            while(start <= end){
                int mid = start + (end - start) / 2;
                if(b[mid] <= it){
                    start = mid + 1;
                }
                else{
                    end = mid - 1;
                }
            }
            ans.push_back(end + 1);
        }
        return ans;
    }
};

File: Day179_2025/Q179_GFG.java
import java.util.*;

class Q179_GFG {
    public static ArrayList<Integer> countLessEq(int[] a, int[] b) {
        Arrays.sort(b);
        ArrayList<Integer> ans = new ArrayList<>();

        for (int num : a) {
            int start = 0, end = b.length - 1;
            while (start <= end) {
                int mid = start + (end - start) / 2;
                if (b[mid] <= num) {
                    start = mid + 1;
                } else {
                    end = mid - 1;
                }
            }
            ans.add(end + 1); // Because indices start at 0
        }

        return ans;
    }
}
